src/sqlfluff/core/rules/base.py

```diff
@@ -1067,11 +1077,45 @@ def eval(self, **kwargs):
                     code, self.name
                 )
             )
-        self._register[code] = dict(code=code, description=description, cls=cls)
+
+        try:
+            assert (
+                "all" in cls.groups
+            ), "Rule {!r} must belong to the 'all' group".format(code)
+            groups = cls.groups
+        except AttributeError:
+
+            raise AttributeError(
+                (
+                    "Rule {!r} doesn't belong to any rule groups. "
+                    "All rules must belong to at least one group"
+                ).format(code)
+            )
+
+        self._register[code] = dict(
+            code=code, description=description, groups=groups, cls=cls
+        )
 
         # Make sure we actually return the original class
         return cls
 
+    def _expand_config_rule_group_list(
+        self, rule_list: List[str], valid_groups: Set[str]
+    ) -> List[str]:
+        expanded_rule_list: List[str] = []
+        for r in rule_list:
+            if r in valid_groups:
+                rules_in_group = [
+                    rule
+                    for rule, rule_dict in self._register.items()
+                    if r in rule_dict["groups"]
+                ]
```

Comment:
Maybe a question for a future PR: Does this support nested rule groups, e.g., maybe later we want to define the core group as the combination of several groups, e.g. whitespace, "making things explicit", "foobar". In that case, expansion of groups would need to be recursive.

Reply:
I could see that as a nice DRY alternative, rather than manually specifying the same rule in different groups over and over again. I agree though that we should leave it as a consideration for a future PR -- who knows, maybe no one will care about rule groups 😂